homeassistant/components/axis/switch.py

```diff
@@ -1,46 +1,88 @@
 """Support for Axis switches."""
 
+from collections.abc import Callable, Iterable
+from dataclasses import dataclass
+from functools import partial
 from typing import Any
 
 from axis.models.event import Event, EventOperation, EventTopic
 
-from homeassistant.components.switch import SwitchEntity
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .entity import AxisEventEntity
 from .hub import AxisHub
 
 
+@dataclass(frozen=True, kw_only=True)
+class AxisSwitchDescription(SwitchEntityDescription):
+    """Axis switch entity description."""
+
+    event_topic: EventTopic
+    """Event topic that provides state updates."""
+    name_fn: Callable[[AxisHub, Event], str]
+    """Function providing the corresponding name to the event ID."""
+    supported_fn: Callable[[AxisHub, Event], bool]
+    """Function validating if event is supported."""
+
+
+ENTITY_DESCRIPTIONS = (
+    AxisSwitchDescription(
+        key="Relay state control",
+        device_class=SwitchDeviceClass.OUTLET,
+        entity_category=EntityCategory.CONFIG,
+        event_topic=EventTopic.RELAY,
+        supported_fn=lambda hub, event: isinstance(int(event.id), int),
+        name_fn=lambda hub, event: hub.api.vapix.ports[event.id].name,
```

Comment:
What is the name of a switch? Can we slowly start migrating for has entity name = True?

Reply:
`_attr_has_entity_name = True` in the base class AxisEntity, name is the specific port information `{SWITCH_DOMAIN}.{NAME}_{PORT NAME}`